stdlib/list.ml

```diff
@@ -56,14 +56,18 @@ let rec rev_append l1 l2 =
 
 let rev l = rev_append l []
 
-let rec init_aux acc n f =
-  if n < 0 then acc else
-  init_aux ((f n)::acc) (n-1) f
+let rec init_tailrec_aux acc i n f =
+  if i >= n then acc
+  else init_tailrec_aux ((f i)::acc) (i+1) (n) f
+
+let rec init_aux i n f =
+  if i >= n then []
+  else (f i)::init_aux (i+1) n f
```

Comment:
Without explicit sequencing (`let fi = f i in...` in my message) this implementation fails to maintain the evaluation order spec.

In the top level:
```ocaml
# let _ = init_aux 0 5 print_int;;
43210- : unit list = [(); (); (); (); ()]
``` 

to test.